### PR TITLE
jmavsim: document headless mode

### PR DESCRIPTION
**Pull request by julianoes**
@katzfey added the option to run jMAVSim without GUI.

### CODE REVIEW

File: en/simulation/jmavsim.md [hunk 1]
```diff
@@ -121,6 +121,13 @@ This allows a faster testing cycle (restarting jMAVSim takes significantly more
 
 JMAVSim can be used for multi-vehicle simulation: [Multi-Vehicle Sim with JMAVSim](../simulation/multi_vehicle_jmavsim.md).
 
+## Headless Mode
```
Comment (hamishwillee) [inline en/simulation/jmavsim.md]: You've put it under the wrong heading - needs to go above `Multi-Vehicle Simulation` and the level needs to change from `##` to `###`.

FWIW This is excellent. Should allow me to run JMAVSim in Ubuntu in a VM, which currently fails due to graphics driver issues.

**Comment by hamishwillee** [comment]
Note, I just ran this in my VM and I still get errors that tell me it is trying to do something with the GUI. Am I missing the point?

```
INFO  [mavlink] MAVLink only on localhost (set param MAV_BROADCAST = 1 to enable network)
INFO  [logger] Opened full log file: ./log/2019-08-23/02_11_41.ulg
INFO  [px4] Startup script returned successfully
pxh> 3D [dev] 1.6.0-pre12-daily-experimental daily

com.jogamp.opengl.GLException: X11GLXDrawableFactory - Could not initialize shared resources for X11GraphicsDevice[type .x11, connection :0, unitID 0, handle 0x0, owner false, ResourceToolkitLock[obj 0x34f73f2d, isOwner false, <22be4c38, 413061ad>[count 0, qsz 0, owner <NULL>]]]
	at jogamp.opengl.x11.glx.X11GLXDrawableFactory$SharedResourceImplementation.createSharedResource(X11GLXDrawableFactory.java:326)
	at jogamp.opengl.SharedResourceRunner.run(SharedResourceRunner.java:297)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:688)
	at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:580)
	at jogamp.opengl.x11.glx.X11GLXDrawableFactory$SharedResourceImplementation.createSharedResource(X11GLXDrawableFactory.java:297)
	... 2 more
com.jogamp.opengl.GLException: J3D-Renderer-1: createImpl ARB n/a but required, profile > GL2 requested (OpenGL >= 3.1). Requested: GLProfile[GL3bc/GL3bc.hw], current: 3.0 (Compat profile, compat[ES2], FBO, hardware) - 3.0 Mesa 17.2.8
	at jogamp.opengl.x11.glx.X11GLXContext.createImpl(X11GLXContext.java:418)
	at jogamp.opengl.GLContextImpl.makeCurrentWithinLock(GLContextImpl.java:759)
	at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:642)
	at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:580)
	at javax.media.j3d.JoglPipeline$QueryCanvas.doQuery(JoglPipeline.java:8615)
	at javax.media.j3d.JoglPipeline$QueryCanvas.access$100(JoglPipeline.java:8566)
	at javax.media.j3d.JoglPipeline.createQueryContext(JoglPipeline.java:6562)
	at javax.media.j3d.Canvas3D.createQueryContext(Canvas3D.java:4609)
	at javax.media.j3d.Canvas3D.createQueryContext(Canvas3D.java:3606)
	at javax.media.j3d.Renderer.doWork(Renderer.java:461)
	at javax.media.j3d.J3dThread.run(J3dThread.java:271)

DefaultRenderingErrorListener.errorOccurred:
CONTEXT_CREATION_ERROR: Renderer: Error creating Canvas3D graphics context for queryProperties()
graphicsDevice = X11GraphicsDevice[screen=0]
canvas = me.drton.jmavsim.Visualizer3D$CustomCanvas3D[canvas0,0,0,0x0,invalid]
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.jdt.internal.jarinjarloader.JarRsrcLoader.main(JarRsrcLoader.java:61)
Caused by: java.lang.IllegalStateException: Canvas3D: Non-recoverable graphics configuration error
	at javax.media.j3d.Canvas3D.queryProperties(Canvas3D.java:3595)
	at me.drton.jmavsim.Visualizer3D.createEnvironment(Visualizer3D.java:269)
	at me.drton.jmavsim.Visualizer3D.<init>(Visualizer3D.java:215)
	at me.drton.jmavsim.Simulator.<init>(Simulator.java:183)
	at me.drton.jmavsim.Simulator.main(Simulator.java:913)
	... 5 more
X11Util.Display: Shutdown (JVM shutdown: true, open (no close attempt): 2/2, reusable (open, marked uncloseable): 0, pending (open in creation order): 2)
X11Util: Open X11 Display Connections: 2
X11Util: Open[0]: NamedX11Display[:0, 0x7f7624000a50, refCount 1, unCloseable false]
X11Util: Open[1]: NamedX11Display[:0, 0x7f763c0267a0, refCount 1, unCloseable false]
```

**Comment by julianoes** [comment]
@hamishwillee did you use this one? https://github.com/PX4/Firmware/pull/12781

File: en/simulation/jmavsim.md (hunk 1 above)
Comment (julianoes) [reply en/simulation/jmavsim.md]: Fixed and force-pushed.

**Comment by hamishwillee** [comment]
Ah no, thought this is already in master!

**Comment by julianoes** [comment]
Sorry, yes. I should have linked that.

**Comment by hamishwillee** [comment]
Thanks, awesome!

@julianoes That merged and I retested. Yes, the GUI stuff is properly removed and this does run :-)

Note, I still get lots of problems running in VM in practise. Given that Gazebo is much more heavyweight and does not have this problem, could I consider the following sensor warnings/errors as a bug?

```
INFO  [ecl/EKF] EKF aligned, (pressure height, IMU buf: 22, OBS buf: 14)
INFO  [ecl/EKF] EKF GPS checks passed (WGS-84 origin set)
ERROR [sensors] Accel #0 fail:  TIMEOUT!
ERROR [sensors] Sensor Accel #0 failed. Reconfiguring sensor priorities.
WARN  [sensors] Remaining sensors after failover event 0: Accel #0 priority: 1
ERROR [sensors] Accel #0 fail:  TIMEOUT!
ERROR [sensors] Sensor Accel #0 failed. Reconfiguring sensor priorities.
WARN  [sensors] Remaining sensors after failover event 0: Accel #0 priority: 1
ERROR [sensors] Accel #0 fail:  TIMEOUT!
ERROR [sensors] Sensor Accel #0 failed. Reconfiguring sensor priorities.
WARN  [sensors] Remaining sensors after failover event 0: Accel #0 priority: 1
INFO  [ecl/EKF] EKF commencing GPS fusion
ERROR [sensors] Accel #0 fail:  TIMEOUT!
ERROR [sensors] Sensor Accel #0 failed. Reconfiguring sensor priorities.
WARN  [sensors] Remaining sensors after failover event 0: Accel #0 priority: 1
ERROR [sensors] Accel #0 fail:  TIMEOUT!
ERROR [sensors] Sensor Accel #0 failed. Reconfiguring sensor priorities.
WARN  [sensors] Remaining sensors after failover event 0: Accel #0 priority: 1
```

**Comment by julianoes** [comment]
What sort of system and what VM is that? I almost never see that in practice.

**Comment by hamishwillee** [comment]
This is Ubuntu 16.04 LTS running on VMWare Workstation Pro 14, on a Windows Alienware 15R2 Intel core i7 with 16GB RAM from an SSD drive, where VM has 8GB ram and 2 processors. Enough grunt to run Gazebo headless OK. 2.5 years old

**Comment by julianoes** [comment]
That should indeed be powerful enough.

**Comment by hamishwillee** [comment]
I'll try on a few other systems and see if it is reproducable.

**Comment by hamishwillee** [comment]
@julianoes FYI it is something to do with VMWare. I tried same setup on VirtualBox and everything works fine (even with only 1 core on VM it doesn't pop up same errors, albeit everything is pretty slow)

**Comment by hamishwillee** [comment]
Actually, I still get the timeout on VirtualBox, but it is just not quite so quick to appear.
image

**Pull request merged**